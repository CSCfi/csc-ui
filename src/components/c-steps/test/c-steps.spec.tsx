import { newSpecPage } from '@stencil/core/testing';
import { CSteps } from '../c-steps';
import { CStep } from '../c-step';

global.ResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}));

describe('c-steps', () => {
  it('renders default html', async () => {
    const page = await newSpecPage({
      components: [CSteps],
      html: `<c-steps></c-steps>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('renders steps', async () => {
    const page = await newSpecPage({
      components: [CSteps, CStep],
      html: `
      <c-steps>
        <c-step></c-step>
        <c-step></c-step>
        <c-step></c-step>
      </c-steps>
      `,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('renders steps with value of current step', async () => {
    const page = await newSpecPage({
      components: [CSteps, CStep],
      html: `
      <c-steps value="2">
        <c-step></c-step>
        <c-step></c-step>
        <c-step></c-step>
      </c-steps>
      `,
    });

    expect(page.root).toMatchSnapshot();
  });
});
