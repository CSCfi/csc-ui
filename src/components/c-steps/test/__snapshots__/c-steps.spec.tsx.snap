// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-steps renders default html 1`] = `
<c-steps tabindex="0">
  <mock:shadow-root>
    <div aria-hidden="true" class="c-steps">
      <slot></slot>
    </div>
  </mock:shadow-root>
</c-steps>
`;

exports[`c-steps renders steps 1`] = `
<c-steps tabindex="0">
  <mock:shadow-root>
    <div aria-hidden="true" class="c-steps">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <c-step>
    <mock:shadow-root>
      <div class="c-step">
        <div class="c-step__indicator">
          <div class="dot"></div>
        </div>
        <div class="c-step__label">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </c-step>
  <c-step>
    <mock:shadow-root>
      <div class="c-step">
        <div class="c-step__indicator">
          <div class="dot"></div>
        </div>
        <div class="c-step__label">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </c-step>
  <c-step>
    <mock:shadow-root>
      <div class="c-step">
        <div class="c-step__indicator">
          <div class="dot"></div>
        </div>
        <div class="c-step__label">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </c-step>
</c-steps>
`;

exports[`c-steps renders steps with value of current step 1`] = `
<c-steps tabindex="0" value="2">
  <mock:shadow-root>
    <div aria-hidden="true" class="c-steps">
      <slot></slot>
    </div>
  </mock:shadow-root>
  <c-step>
    <mock:shadow-root>
      <div class="c-step c-step--complete">
        <div class="c-step__indicator">
          <div class="complete">
            <svg viewBox="0 0 100 100">
              <path class="path" d="M 12 52 l 24 24 l 47 -47 l -3 -3 l -44 44 l -21 -21 l -3 3"></path>
            </svg>
          </div>
        </div>
        <div class="c-step__label">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </c-step>
  <c-step>
    <mock:shadow-root>
      <div class="c-step">
        <div class="c-step__indicator">
          <div class="current dot"></div>
        </div>
        <div class="c-step__label">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </c-step>
  <c-step>
    <mock:shadow-root>
      <div class="c-step">
        <div class="c-step__indicator">
          <div class="dot"></div>
        </div>
        <div class="c-step__label">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </c-step>
</c-steps>
`;
